atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/ValidatingTransactionScopedCache.java

```diff
@@ -101,6 +109,38 @@ public void delete(TableReference tableReference, Set<Cell> cells) {
         }
     }
 
+    @Override
+    public NavigableMap<byte[], RowResult<byte[]>> getRows(
+            TableReference tableRef,
+            Iterable<byte[]> rows,
+            ColumnSelection columnSelection,
+            Function<Set<Cell>, Map<Cell, byte[]>> cellLoader,
+            Function<Iterable<byte[]>, NavigableMap<byte[], RowResult<byte[]>>> rowLoader) {
+        if (shouldValidate()) {
+            NavigableMap<byte[], RowResult<byte[]>> remoteReads = rowLoader.apply(rows);
+            NavigableMap<byte[], RowResult<byte[]>> cacheReads = delegate.getRows(
+                    tableRef,
+                    rows,
+                    columnSelection,
+                    cellsToRead -> KeyedStream.of(cellsToRead)
+                            .map(cell -> remoteReads
+                                    .get(cell.getRowName())
+                                    .getColumns()
+                                    .get(cell.getColumnName()))
+                            .filter(Objects::nonNull)
+                            .collectToMap(),
+                    rowsToRead -> {
+                        SortedSet<byte[]> toReadNavigable = new TreeSet<>(UnsignedBytes.lexicographicalComparator());
+                        rowsToRead.forEach(toReadNavigable::add);
```

Comment:
Why do we need SortedSet?

Reply:
Sorted set uses ordering for lookups, so it will not break on `byte[]` with different references